lib/amqp/session.rb

```diff
@@ -930,9 +930,15 @@ def receive_frame(frame)
         begin
           receive_frameset(@frames[frame.channel])
         ensure
-          # for channel.close, frame.channel will be nil. MK.
-          clear_frames_on(frame.channel) if @frames[frame.channel]
+          begin
+            clear_frames_on(frame.channel) if @frames[frame.channel] # for channel.close, frame.channel will be nil. MK.
+          rescue Exception => ex
```

Comment:
Are you positive you want `Exception`? That will rescue script errors and other major problems. I was thinking an `ensure_safe` might work which doesn't rescue at that low a level.